kafka_utils/util/utils.py

```diff
@@ -34,9 +34,48 @@ def get_module(module_full_name):
         return importlib.import_module(module_full_name)
 
 
+def child_class(class_types, base_class):
+    """
+    Find the child-most class of `base_class`.
+
+    Examples:
+        class A:
+            pass
+
+        class B(A):
+            pass
+
+        class C(B):
+            pass
+
+        child_class([A, B, C], A) = C
+    """
+    subclasses = set()
+    for class_type in class_types:
+        if class_type is base_class:
+            continue
+        if issubclass(class_type, base_class):
+            subclasses.add(class_type)
+
+    if len(subclasses) == 0:
+        return None
+    elif len(subclasses) == 1:
+        return subclasses.pop()
+    else:
+        # If more than one class is a subclass of `base_class`
+        # It is possible that one or more classes are subclasses of another
+        # class (see example above).
+        # Recursively find the child-most class.
+        for c in subclasses:
+            child = child_class(subclasses, c)
+            if child is not None:
+                return child
+
```

Comment:
I am not sure how this function is used but shouldnt we be returning `subclasses.pop()` here in case we are not able to find the most-child subclass recursively. This might happen in the following case
`class A class B(A) class C(A)` where `B` and `C` are not subclasses of each other. I am not sure whether this case is possible, so this is more of a question 😄

Reply:
Good point!

Reply:
It shouldn't happen (otherwise we'd get a somewhat undefined behaviour), but we "solve" this by making sure only one child-most class exists per module. May as well return at least one result in case that happens.